src/clr-angular/data/datagrid/_datagrid.clarity.scss

```diff
@@ -915,9 +915,174 @@
   }
 
   /*
-        The following classes are used by the renderer when performing internal operations.
-        Using the browser calculations improves rendering performance.
-    */
+   * The following are styles for Detail Pane
+   */
+  /* autoprefixer grid: autoplace */
+  .datagrid-host {
+    display: grid;
+    grid-template-columns: 1fr;
+    grid-template-rows: auto auto 1.5rem;
+    grid-template-areas:
+      'datagrid-actions'
+      'datagrid'
+      'datagrid-footer';
+  }
+  .datagrid-host.datagrid-detail-open {
+    grid-template-columns: 1fr 2fr;
+    grid-template-rows: auto auto 1.5rem;
+    grid-template-areas:
+      'datagrid-actions datagrid-actions'
+      'datagrid datagrid-detail'
+      'datagrid-footer datagrid-detail';
+  }
+  .datagrid {
+    grid-area: datagrid;
+    width: auto;
+  }
+  .datagrid-action-bar {
+    grid-area: datagrid-actions;
+  }
+  .datagrid-footer {
+    grid-area: datagrid-footer;
+  }
+  .datagrid-detail-pane {
+    grid-area: datagrid-detail;
+  }
+
+  .datagrid-detail-open {
+    .datagrid {
+      border-top-right-radius: 0;
+      border-right: none;
+    }
+    .datagrid-row {
+      border-right: 1px solid $clr-table-border-color;
+    }
+    .datagrid-footer {
+      border-bottom-right-radius: 0;
+    }
+    .pagination {
+      width: 100%;
+    }
+    .pagination-description-compact {
+      text-align: left;
+      flex: 1;
+    }
+    .datagrid-footer .pagination-list {
+      margin-right: 0;
+    }
+  }
+
+  .datagrid-row-detail-open {
+    position: relative;
+
+    &:before {
+      content: '';
+      display: inline-block;
+      position: absolute;
+      border: 10px solid $clr-table-border-color;
+      border-color: transparent $clr-table-border-color transparent transparent;
+      top: calc(0.5rem - 2px);
+      right: 0;
+    }
+
+    &:after {
+      content: '';
+      display: inline-block;
+      position: absolute;
+      border: 9px solid $clr-white;
+      border-color: transparent $clr-white transparent transparent;
+      top: calc(0.5rem - 1px);
+      right: -1px;
+    }
+  }
+
+  .datagrid-detail-pane {
+    margin-top: 0.5rem;
+    border: $clr-default-borderwidth solid $clr-datagrid-popover-border-color;
+    border-left: none;
+    border-top-right-radius: $clr-global-borderradius;
+    border-bottom-right-radius: $clr-global-borderradius;
+    background: $clr-white;
+    overflow: hidden;
+  }
+
+  .datagrid-detail-pane-content {
+    display: flex;
+    flex-direction: column;
+    overflow: scroll;
+    height: 100%;
+  }
+
+  .datagrid-detail-body {
+    flex: 1 1 auto;
+  }
+
+  .datagrid-detail-header {
+    flex: 0 0 auto;
+    font-size: 0.75rem;
+    line-height: 1.5rem;
+    padding-left: 1rem;
+    margin-top: 0;
+    display: flex;
+    flex-direction: row;
+
+    .datagrid-detail-header-title {
+      flex: 1 1 auto;
+    }
+
+    .datagrid-detail-pane-close {
+      flex: 1 1 4rem;
+      padding: 0;
+      cursor: pointer;
+
```

Comment:
Does this need something like:
```
    display: flex;
    justify-content: flex-end;
```

to keep the close button on the right side of its container?